// ng2-adsense/esm2015/adsense.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ADSENSE_TOKEN } from './adsense-config';
import { AdsenseComponent } from './adsense.component';
import * as ɵngcc0 from '@angular/core';
export class AdsenseModule {
    static forRoot(config = {}) {
        return {
            ngModule: AdsenseModule,
            providers: [{ provide: ADSENSE_TOKEN, useValue: config }],
        };
    }
}
AdsenseModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AdsenseModule });
AdsenseModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AdsenseModule_Factory(t) { return new (t || AdsenseModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AdsenseModule, { declarations: function () { return [AdsenseComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [AdsenseComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AdsenseModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [AdsenseComponent],
                declarations: [AdsenseComponent]
            }]
    }], null, null); })();
